eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Mempool.hs

```diff
@@ -183,7 +183,25 @@ class
           . left ApplyTxError
           $ res
 
-instance PraosCrypto c => ApplyTx (ShelleyEra c)
+instance
+  ( CC.Crypto c,
+    Eq (PredicateFailure (Core.EraRule "LEDGER" (ShelleyEra c))),
+    Signal (Core.EraRule "LEDGER" (ShelleyEra c)) ~ Shelley.Tx (ShelleyEra c),
+    State (Core.EraRule "LEDGER" (ShelleyEra c))
+      ~ (LedgerState.UTxOState (ShelleyEra c), LedgerState.DPState c),
+    Environment (Core.EraRule "LEDGER" (ShelleyEra c)) ~ LedgerEnv (ShelleyEra c),
+    BaseM (Core.EraRule "LEDGER" (ShelleyEra c)) ~ ShelleyBase,
+    PredicateFailure
+      (Core.EraRule "LEDGER" (ShelleyEra c))
+      ~ LedgerPredicateFailure (ShelleyEra c),
+    STS (Core.EraRule "LEDGER" (ShelleyEra c)),
+    FromCBOR (PredicateFailure (Core.EraRule "DELEGS" (ShelleyEra c))),
+    Show (PredicateFailure (Core.EraRule "DELEGS" (ShelleyEra c))),
+    FromCBOR (PredicateFailure (Core.EraRule "UTXOW" (ShelleyEra c))),
+    ToCBOR (PredicateFailure (Core.EraRule "DELEGS" (ShelleyEra c))),
+    ToCBOR (PredicateFailure (Core.EraRule "UTXOW" (ShelleyEra c)))
+  ) =>
+  ApplyTx (ShelleyEra c)
```

Comment:
Do we like the `ApplyTx` remaining here, but changing the `PraosCrypto` constraint? I would love to understand why removing `PraosCrypto` forced all the constraints above, they do not seem to be covered by `PraosCrypto`.

Reply:
Odd